### PR TITLE
B 19888 estimate price int

### PR DESCRIPTION
**Pull request by traskowskycaci**
Do not merge before #12811 

## Summary

When the prime provides an estimated weight, we want to calculate the estimated price and store it on the service item in the `pricing_estimate` column.

To test:
1. Run `make tasks_save_ghc_fuel_price_data`
2. Create a shipment in the prime sim, and make a note of your requested pickup date and your estimated weight.
3. Approve the shipment as the TOO.
4. Check the mto_service_items table and verify that prices exist for the initial 6 service items that get created upon TOO approval.
5. Go back to the prime sim
6. Update the shipment to have the same `actual` pickup and `actual` weight that you put down for the requested pickup and estimated weight.
7. After updating, create a payment request as the prime sim.
8. Make sure you use the shipment you created and updated - then add all the service items and submit the payment request.
9. As the TIO, review the payment request and compare the prices on that page to the values in the database.

A note on the tests - there is quite a bit of test setup data required to have all the inputs for every pricing function. The pricing functions I utilized already have pretty extensive tests written for them, so I don't see a need to retest them in the shipment approver tests.

image

image

### CODE REVIEW

File: pkg/services/mto_service_item/mto_service_item_creator.go [hunk 1]
```diff
@@ -24,10 +26,247 @@ type createMTOServiceItemQueryBuilder interface {
...
+		Where("re_contracts.code = ?", contractCode).
+		First(&domesticServiceArea)
+	if err != nil {
+		return domesticServiceArea, fmt.Errorf("unable to find domestic service area for %s under contract code %s", zip3, contractCode)
+	}
+
+	return domesticServiceArea, nil
+}
+
+const weightBasedDistanceMultiplierLevelOne = "0.000417"
```
Comment (deandreJones) [inline pkg/services/mto_service_item/mto_service_item_creator.go:234]: where did these numbers come from?

Comment (traskowskycaci) [reply pkg/services/mto_service_item/mto_service_item_creator.go:234]: pkg/payment_request/service_param_value_lookups/fsc_weight_based_distance_multiplier_lookup.go

**Review by TevinAdams — APPROVED** [review]
Test case worked. Prices on payment request matched what what was displayed in the db as the pricing estimate value.

Reviewed code and tried to make things break. Looks good.

**Review by deandreJones — APPROVED** [review]

**Review by danieljordan-caci — APPROVED** [review]
Tested locally, all seems to work as intended per your test cases. Code looks good and seems like you're handling errors appropriately. LGTM.

**Review by MichaelEngelbertCACI — APPROVED** [review]
Checked out and tested locally, looks good on my end.

**Comment by traskowskycaci** [comment]
Waiting to merge this one until https://github.com/transcom/mymove/pull/12811 makes it into main. Both PRs will fail on client_test until fix for I-12947 is in.

**Comment by traskowskycaci** [comment]
> Waiting to merge this one until #12811 makes it into main. Both PRs will fail on client_test until fix for I-12947 is in.

#12811 has been merged in. Doing a final test of functionality on this branch then will merge

**Pull request merged**